File: source/java/src/main/java/com/aerospike/examples/Parameter.java
package com.aerospike.examples;

public class Parameter<T> {
    private final String name;
    private final String description;
    private T value;
    
    public Parameter(String name, T value) {
        this(name, value, null);
    }
    
    public Parameter(String name, T value, String description) {
        this.name = name;
        this.value = value;
        this.description = description;
    }

    public T get() {
        return value;
    }
    
    public String getName() {
        return name;
    }
    
    public String getDescription() {
        return description;
    }
}

File: source/python-sdk/usecasecookbook/parameter.py
"""Mirrors ../../java-sdk's Parameter<T> - a simple mutable named value that use cases
expose via UseCase.get_params() so users can tune things like record counts without
editing code. Unlike the Java version, ``value`` is just a public attribute - Python has
no need for the reflection trick the Java UseCaseExecutor uses to mutate a "private" field.
"""

from dataclasses import dataclass
from typing import Generic, Optional, TypeVar

T = TypeVar("T")


@dataclass
class Parameter(Generic[T]):
    name: str
    value: T
    description: Optional[str] = None

    def get(self) -> T:
        return self.value
